velox/benchmarks/tpch/TpchBenchmark.cpp

```diff
@@ -127,15 +136,31 @@ BENCHMARK(q18) {
 
 int main(int argc, char** argv) {
   folly::init(&argc, &argv, false);
+  bool runQueryVerboseSet = isFlagSet("run_query_verbose");
+  if (!runQueryVerboseSet && isFlagSet("include_custom_stats")) {
+    std::cout
+        << "--include_custom_stats must be specified only with --run_query_verbose"
+        << std::endl;
+    exit(1);
+  }
   benchmark.initialize();
   queryBuilder =
       std::make_shared<TpchQueryBuilder>(toFileFormat(FLAGS_data_format));
   queryBuilder->initialize(FLAGS_data_path);
-  if (FLAGS_run_query_verbose == -1) {
+  if (!runQueryVerboseSet) {
     folly::runBenchmarks();
   } else {
     const auto queryPlan = queryBuilder->getQueryPlan(FLAGS_run_query_verbose);
     const auto task = benchmark.run(queryPlan);
+    if (not task->isFinished()) {
```

Comment:
Should we fold this logic into readCursor? Alternatively, perhaps, add a helper function to QueryAssertions.h to wait for task to complete.

Reply:
`assertQuery` in `QueryAssertions.h` already uses this. Adding a helper function will be nice. Will do that.

Reply:
I think we can use TaskStats.executionEndTimeMs - TaskStats.executionStartTimeMs .